src/state_machine/signer/mod.rs

```diff
@@ -61,6 +63,24 @@ pub enum Error {
     /// An encryption error occurred
     #[error("Encryption error: {0}")]
     Encryption(#[from] EncryptionError),
+    /// An AES-GCM error occurred
+    #[error("AES-GCM: {0}")]
+    AesGcm(AesGcmError),
+    #[error("integer conversion error")]
+    /// An error during integer conversion operations
+    TryFromInt,
+}
+
+impl From<AesGcmError> for Error {
```

Comment:
I think you can replace this with AesGcm(#[from] AesGcmError) above no?

Reply:
Yes, we should.

Reply:
No, we can't:
```
83 |     AesGcm(#[from] AesGcmError),
   |              ^^^^ method cannot be called on `&Error` due to unsatisfied trait bounds
   |
  ::: /Users/axoloki/.cargo/registry/src/index.crates.io-6f17d22bba15001f/aead-0.5.2/src/lib.rs:79:1
   |
79 | pub struct Error;
   | ---------------- doesn't satisfy `aes_gcm::Error: AsDynError<'_>` or `aes_gcm::Error: StdError`
   |
   = note: the following trait bounds were not satisfied:
           `aes_gcm::Error: StdError`
           which is required by `aes_gcm::Error: AsDynError<'_>`
           `&aes_gcm::Error: StdError`
           which is required by `&aes_gcm::Error: AsDynError<'_>`
```
`thiserror` is awesome, but sadly we can't do much about upstream that errors that don't satisfy the trait bounds.

Reply:
Fortunately, the AES DOS fix required me to define an `EncryptionError` type, which takes care of wrapping `aes_gcm::Error`.  So we only have to do the manual chaining once.